### ISSUE TITLE
Context can't be evaluated in components that have been removed from the tree

### ISSUE THREAD

**Issue by Tiagoperes** [ZupIT/beagle-web-core#257]
By the title one might think this behavior is expected, but take the following situation:

Suppose a list, where there's a remove button in each item, and the remove button does the following:
1. Immediately removes the item from the list (optimistic UI)
2. Sends a request to the server so the item can be also removed from the database
3. In case of error, rolls back the list to the previous state, i.e. reinsert the removed item back into the list.

The action attached to the event `onRemove` that would represent this behavior could be the following:

```typescript
onRemove: [
    {
      _beagleAction_: 'beagle:setContext',
      contextId: 'data',
      path: 'notes',
      value: '@{remove(data.notes, item)}',
    },
    {
      _beagleAction_: 'beagle:sendRequest',
      url: `https://www.myapp.com/list/@{item.id}`,
      method: 'DELETE',
      onError: [
        {
          _beagleAction_: 'beagle:setContext',
          contextId: 'data',
          path: 'notes',
          value: '@{insert(data.notes, item, index)}',
        },
      ],
    },
  ]
```

In the  code above, the array to fill the list is located in the context with id `data` (`data.notes`).

The error case (rolling back the list state) wont't work at all. Instead, when we get the error case, the following message will be printed to the console: `Could not find context with id "data" for element of type {elementType} and id {elementId}`.

This happens because by the time the action `setContext` ( at `sendRequest: onError`) is executed, the item has already been removed and we can't parse `@{data.notes}`, because no context is linked to a component that doesn't even exist.

Examples in playground: https://beagle-playground-beta.netlify.app/#/cloud/ff6f1af2759e4b0d8179f1f30f658533

How could we fix this? Do we need to fix it? How would the other platforms behave?

**Comment by Tiagoperes** [comment]
Remove the "skip" from the test at `__tests__/integration/beagle-keep/assertions/actions/home/notes.remove.spec.ts`

**Comment by Tiagoperes** [comment]
As discussed in the daily of today, we decided this behavior is expected.

Instead of removing the element, we should actually hide it, and then, just upon a successful response we should remove it.

Today, we don't have a native way to hide an element in Beagle, the developer must create a custom component for this. For this reason, we should give priority to this issue https://github.com/ZupIT/beagle/issues/794

As a task to close this issue, we should change the error message so the problem becomes clear to the developer. Something like: "you attempted to access a context relative to an element that has been removed from the tree. This is not possible".

A bigger discussion on this would be: should we abort every action triggered by components that don't exist anymore?
